### PR TITLE
feat(logs): filterName property in MetricFilter

### PR DESCRIPTION
**Pull request by vinayak-kukreja**
This PR is recreating and adding to https://github.com/robertd PR: https://github.com/aws/aws-cdk/pull/23053

Here, `filterName` property is being added for `MetricFilter`
And, also adding an existing integ test: `integ.expose-metric.ts` with the `IntegTest` construct.

Closes https://github.com/aws/aws-cdk/issues/17626

Co-authored-by: Robert Djurasaj(https://github.com/robertd)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk-testing/framework-integ/test/aws-logs/test/integ.metricfilter-unit.ts.snapshot/manifest.json [hunk 1]
```diff
@@ -0,0 +1,117 @@
+{
```
Comment (corymhall) [inline packages/@aws-cdk-testing/framework-integ/test/aws-logs/test/integ.metricfilter-unit.ts.snapshot/manifest.json:1]: It looks like these are all `ts` snapshots, probably a result of running
integ-runner with `--language typescript`. Can you remove these? We probably
should add a gitignore rule for this as well.

**Review by corymhall — CHANGES_REQUESTED** [review]
Just 2 minor things

**Comment by robertd** [comment]
> You can also add robertd@ as a co-author to the merge commit (if you're reusing his code) but adding `co-authored-by:` in the PR description: https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors.
> 
> We do this on occasion in cdk: https://github.com/aws/aws-cdk/pulls?q=is%3Apr+co-authored-by%3A+is%3Aclosed

Appreciate the love @kaizencc and @vinayak-kukreja 🫶

**Comment by robertd** [comment]
Side topic… I wasn’t aware `co-authored-by:`, does it take @username as an argument?

**Comment by vinayak-kukreja** [comment]
> Side topic… I wasn’t aware `co-authored-by:`, does it take @username as an argument?

I just followed the format: `Co-authored-by: NAME <NAME@EXAMPLE.COM>`. So adding the name and the email :)

**Comment by robertd** [comment]
You learn something new every day :). Thanks for leading the charge on this PR @vinayak-kukreja 🙌

**Comment by vinayak-kukreja** [comment]
Thank you for your initial contribution @robertd. :)

**Review by corymhall — CHANGES_REQUESTED** [review]
@vinayak-kukreja sorry I missed one thing on my first pass.

File: packages/aws-cdk-lib/aws-logs/lib/metric-filter.ts [hunk 2]
```diff
@@ -42,6 +43,7 @@ export class MetricFilter extends Resource {
     // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-logs-metricfilter.html
     new CfnMetricFilter(this, 'Resource', {
       logGroupName: props.logGroup.logGroupName,
+      filterName: props.filterName,
```
[the remark this answers is not in the document]
Comment (vinayak-kukreja) [reply packages/aws-cdk-lib/aws-logs/lib/metric-filter.ts:46]: No worries, will make the update.

**Review by corymhall — APPROVED** [review]

**Pull request merged**